api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/mchms/DualHIVSyphilisCalculation.java

```diff
@@ -0,0 +1,140 @@
+
+        for (Integer ptId : aliveAndFemale) {
+            boolean eligibleDualHivSyphilisFlag = false;
+            PatientService patientService = Context.getPatientService();
+            Patient patient = patientService.getPatient(ptId);
+
+            Form antenatalVisitForm = MetadataUtils.existing(Form.class, MchMetadata._Form.MCHMS_ANTENATAL_VISIT);
+            EncounterType mchConsultationEncounterType = MetadataUtils.existing(EncounterType.class, MchMetadata._EncounterType.MCHMS_CONSULTATION);
+            Encounter lastANCEnc = EmrUtils.lastEncounter(patient,mchConsultationEncounterType,antenatalVisitForm );
+            Obs pregStatusObs = EmrCalculationUtils.obsResultForPatient(pregStatusObss, ptId);
+
+            Obs hepBObs = EmrCalculationUtils.obsResultForPatient(lastHepatitisTesting, ptId);
+            Obs syphilsObs = EmrCalculationUtils.obsResultForPatient(lastSyphilisTesting, ptId);
+            Obs resultOfHivTestingObs = EmrCalculationUtils.obsResultForPatient(resultOfHivTesting, ptId);
+            Obs lmpObs = EmrCalculationUtils.obsResultForPatient(lmp, ptId);
+
+            lmpDate = lmpObs.getObsDatetime();
+
+            if(hepBObs !=null){
+                if (hepBObs.getObsDatetime() !=null){
+                    hepatitisTestDate = hepBObs.getObsDatetime();
+                    hepBTestMonths = monthsBetween(hepatitisTestDate, lmpDate);
+                }
+            }
+            if (syphilsObs != null) {
+                if (syphilsObs.getObsDatetime() != null) {
+                    syphilisTestDate = syphilsObs.getObsDatetime();
+                    syphilisTestMonths = monthsBetween(syphilisTestDate, lmpDate);
+                }
+            }
+
+            if(resultOfHivTestingObs !=null) {
+                if (resultOfHivTestingObs.getObsDatetime() != null) {
+                    hivTestDate = resultOfHivTestingObs.getObsDatetime();
+                    hivTestMonths = monthsBetween(hivTestDate, lmpDate);
+
+                }
+            }
+
+            int pregnancyMonths = monthsBetween(currentDate, lmpDate);
+            int monthsSinceFirstTrimester = monthsBetween(lmpDate, currentDate);
+            boolean syphilisQuestion = lastANCEnc != null ? EmrUtils.encounterThatPassCodedAnswer(lastANCEnc, syphilisQ, syphilisNegative) : false;
+
+            if(pregStatusObs != null && pregStatusObs.getValueCoded().equals(yes)){
+               if(lastANCEnc !=null) {
+
+                   if (pregnancyMonths >= 1 && pregnancyMonths <= 3) {
+                       if (hepBObs ==null && (hepBTestMonths < 1 || hepBTestMonths > 3)) {
+                           eligibleDualHivSyphilisFlag = true;
+                        dualMessage = "Due for HepB";
+                       }
+                       if (syphilsObs ==null && (syphilisTestMonths < 1 || syphilisTestMonths > 3)) {
+                           eligibleDualHivSyphilisFlag = true;
+                           dualMessage = "Due for Syphils";
```

Comment:
What happens to the flag when HepB is already set?

Reply:
Resolved this